magpie/services.py

```diff
@@ -1382,6 +1382,20 @@ def get_path_parts(self):
                 return path_parts[1:]  # remove extra '' added by split
         return path_parts
 
+    def _get_path_prefix_permission(self, path_parts):
+        # type: (List[str]) -> Tuple[Optional[Str], PermissionRequested]
+        cfg = self.get_config()
+        for prefixes, permission in ((cfg["metadata_type"]["prefixes"], Permission.BROWSE),
+                                     (cfg["data_type"]["prefixes"], Permission.READ)):
+            for prefix in prefixes:
```

Comment:
Please use `pattern_prefix` here as in the original code. I think it is important to get the hint that we are dealing with potential patterns here, and not just plain strings, hence why some matching operations are required rather than plain compares.